src/ImageSharp/Processing/Processors/Drawing/DrawImageProcessor{TPixelBg,TPixelFg}.cs

```diff
@@ -96,6 +96,20 @@ protected override void OnFrameApply(ImageFrame<TPixelBg> source)
             top = 0;
         }
 
+        if (left + foregroundRectangle.Width > source.Width)
```

Comment:
```suggestion
        if (left > source.Width - foregroundRectangle.Width)
```
To prevent possible overflow.